solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/common/SolrSearchSinkUtility.java

```diff
@@ -0,0 +1,116 @@
+    String urlString;
+    SolrClient solrClient = null;
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE)) {
+      urlString = "http://" + config.getSolrHost() + "/solr/" + config.getCollectionName();
+      solrClient = new HttpSolrClient(urlString);
+    } else if (config.getSolrMode().equals(SOLR_CLOUD_MODE)) {
+      CloudSolrClient solrCloudClient = new CloudSolrClient(config.getSolrHost());
+      solrCloudClient.setDefaultCollection(config.getCollectionName());
+      solrClient = solrCloudClient;
+    }
+    return solrClient;
+  }
+
+  /**
+   * Validates whether the host entered for Single Node Solr instance is proper or not.
+   */
+  public static void validateSolrConnectionString(SolrSearchSinkConfig config) {
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE) && config.getSolrHost().contains(",")) {
+      throw new IllegalArgumentException(String.format("Multiple hosts '%s' found for Single Node Solr.",
+                                                       config.getSolrHost()));
+    }
+  }
+
+  /**
+   * Verifies whether the Solr server is running or not. Also, validates whether the collection provided by user is
+   * registered with the Solr server or not.
+   */
+  public static void verifySolrConfiguration(SolrSearchSinkConfig config) {
+    SolrClient client = getSolrConnection(config);
+    try {
+      client.ping();
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+        String.format("Server refused connection at '%s'. Please make sure that either the Solr/Zookeeper services " +
+                        "are properly running or the collection '%s' exists in the Solr Server.", config.getSolrHost(),
+                      config.getCollectionName()));
+    }
+  }
+
+  /**
+   * Validates whether the Id field provided by user is present in the input schema or not.
+   *
+   * @param inputSchema
+   */
+  public static void validateIdField(Schema inputSchema, SolrSearchSinkConfig config) {
+    if (inputSchema != null && inputSchema.getField(config.getIdField()) == null) {
+      throw new IllegalArgumentException(
+        String.format("Idfield '%s' does not exist in the input schema %s", config.getIdField(), inputSchema));
+    }
+  }
+
+  /**
+   * Validates whether the CDAP data types coming from the input schema are compatible with the Solr data types or not.
+   *
+   * @param inputSchema
+   */
+  public static void validateInputFieldsDataType(Schema inputSchema) {
+    /* Currently SolrSearch sink plugin supports CDAP primitives types only: BOOLEAN, INT, LONG, FLOAT, DOUBLE and
+    STRING.*/
+    if (inputSchema != null) {
+      for (Schema.Field field : inputSchema.getFields()) {
+        switch (field.getSchema().getType()) {
+          case BOOLEAN:
+          case INT:
+          case LONG:
+          case FLOAT:
+          case DOUBLE:
+          case STRING:
+            break;
+          default:
+            throw new IllegalArgumentException(
```

Comment:
this means nullable fields are not supported? I would think the id field can't be nullable but other fields can be